hw/misc/hermes.c

```diff
@@ -321,9 +329,10 @@ static const struct hermes_bar2 bar2_init = {
         .perform_dma = hermes_h2c_dma,
         .irq_vector = 0,
         .entry = &hermes_h2c_dma_thread,
-        .name = "h2c-dma",
-    },
-    .c2h = {
+        .name = "h2c-dma"
```

Comment:
I think we need a unique `irq_vector` for each channel. It will allow the driver to know which channel has completed. Tables 81 and 85 of [PG195](https://www.xilinx.com/support/documentation/ip_documentation/xdma/v4_1/pg195-pcie-dma.pdf) contain the following:

> The position of the H2C bits always starts at bit 0. The position of the C2H bits is the index above the last H2C index, and therefore depends on the NUM_H2C_CHNL parameter

Also see Figure 9 there for an example.

Reply:
Also, it'd be nice to have unique names for each channel